plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -290,7 +297,15 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         BigQueryClient client = bigQueryClientFactory.create(session);
         log.debug("getColumnHandles(session=%s, tableHandle=%s)", session, tableHandle);
-        return client.getColumns((BigQueryTableHandle) tableHandle).stream()
+
+        BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
+        ImmutableList.Builder<BigQueryColumnHandle> columns = ImmutableList.builder();
+        columns.addAll(client.getColumns(table));
+        if (table.getPartitionType().isPresent() && table.getPartitionType().get() == INGESTION) {
```

Comment:
Why only for ingestion partition type? Because for a time partitioning the column is already part of the table?

Reply:
As far as I know and tried, a time partitioned table doesn't have these pseudo-columns.

Reply:
Indeed, I tried and see same behaviour. Weird choice of BigQuery to not be consistent with other partition types.